indexer/src/executor.rs

```diff
@@ -66,30 +76,33 @@ impl CustomIndexExecutor {
         })
     }
 
-    pub fn handler(&mut self, handler: CustomHandler) -> &mut Self {
+    pub fn register(&mut self, handler: CustomHandler<T>) -> &mut Self {
         info!("Registering new handler: {:?}", handler);
 
         let event = handler.event.clone();
-        match self.events.get_mut(&event) {
-            Some(handles) => {
-                handles.push(handler);
-            }
-            None => {
-                let h = vec![handler];
-                self.events.insert(event, h);
-            }
-        }
+        self.events
+            .entry(event)
+            .or_insert_with(Vec::new)
+            .push(handler);
 
         self
     }
 }
 
-impl Executor for CustomIndexExecutor {
+impl<T: Entity> Executor for CustomIndexExecutor<T>
+where
+    T: Entity,
+{
     fn from_file(_index: &Path) -> IndexerResult<Self> {
         unimplemented!()
     }
 
-    fn trigger_event(&self, event: ReceiptEvent, bytes: Vec<Vec<u8>>) -> IndexerResult<()> {
+    fn trigger_event(
+        &self,
+        event: ReceiptEvent,
+        _bytes: Vec<Vec<u8>>,
+        receipt: Option<Receipt>,
```

Comment:
- I didn't want to change the `trigger_event` signature here because the already-merged WASM implementation expects things to be a certain way, and I'd rather add a few extra/unused params here than change something that was _just_ merged
- `event` is a type used to identify what event occurred, while `receipt` is the actual `Receipt` value that is used to get the `data`